Microsoft.Alm.Authentication/Global.cs

```diff
@@ -74,7 +74,8 @@ private static string DefaultUserAgent
         {
             get
             {
-                var assemblyName = System.Reflection.Assembly.GetEntryAssembly().GetName();
```

Comment:
Actually, `System.Reflection.Assembly.GetEntryAssembly()` will return caller assembly which invoke Microsoft.Alm.Authentication dll.
If you use visualstudio TestExplorer to run MSTest, it will return null.

If i change it to 
`var assembly = typeof(Microsoft.Alm.Authentication.Global).Assembly;`
it will always return "Microsoft.Alm.Authentication" assembly, this will change the behavior currently.

So i think it is better to change it to 
`var assembly = System.Reflection.Assembly.GetEntryAssembly() ?? typeof(Global).Assembly;`